internal/verifier/change_stream.go

```diff
@@ -98,20 +103,35 @@ func (verifier *Verifier) iterateChangeStream(ctx context.Context, cs *mongo.Cha
 		return err
 	}
 
-	readOneChangeEvent := func() (bool, error) {
-		gotEvent := cs.TryNext(ctx)
-		if gotEvent {
+	readAndHandleOneChangeEventBatch := func() (bool, error) {
+		eventsRead := 0
+		for hasEventInBatch := true; hasEventInBatch; hasEventInBatch = cs.RemainingBatchLength() > 0 {
+			gotEvent := cs.TryNext(ctx)
+
+			if !gotEvent {
+				break
+			}
+
 			var changeEvent ParsedEvent
+
 			if err := cs.Decode(&changeEvent); err != nil {
 				return false, errors.Wrap(err, "failed to decode change event")
 			}
-			err := verifier.HandleChangeStreamEvent(ctx, &changeEvent)
+			err := verifier.HandleChangeStreamEvent(&changeEvent)
 			if err != nil {
 				return false, errors.Wrap(err, "failed to handle change event")
 			}
+
+			eventsRead++
+		}
```

Comment:
Should there be error handling for cs.Err() here?

Reply:
I see where you’re breaking out of the loop, but I don’t see anything that reports the error … ?